src/librustdoc/html/static/main.js

```diff
@@ -1658,9 +1715,13 @@ if (!DOMTokenList.prototype.remove) {
             };
             search_input.onpaste = search_input.onchange;
 
-            var selectCrate = document.getElementById('crate-search');
+            var selectCrate = document.getElementById("crate-search");
             if (selectCrate) {
                 selectCrate.onchange = function() {
+                    var savedCrates = getSavedCrates();
+                    const value = selectCrate.value;
+                    savedCrates[getCratesHash()] = value === "All crates" ? undefined : value;
```

Comment:
As above, it seems reasonable that we can store `"All crates"` in local storage as well since it'll simplify the code here a little and ultimately doesn't affect anything presumably?

Reply:
It's just to avoid keeping unused values. Otherwise, every time you go visit a crate, it'll be added to the local storage, which doesn't sound that nice...

Reply:
It affect local storage space (which is surprisingly limited). Like this, we don't store the defaults.

Reply:
Does this work if there's only one crate? It seems like we presumably just don't run this code then? Can we get comments about:
 * the 'no select crate' dialog case
 * the reasoning for why we want to not store anything for the all crates case